cvat/apps/engine/views.py

```diff
@@ -301,34 +305,34 @@ def perform_create(self, serializer, **kwargs):
             resource_type_field_name=None
         ),
         responses={
-            '202': OpenApiResponse(description='Importing has been started'),
+            '202': OpenApiResponse(RqIdSerializer, description='Importing has been started'),
             '400': OpenApiResponse(description='Failed to import dataset'),
             '405': OpenApiResponse(description='Format is not available'),
         })
     @action(detail=True, methods=['GET', 'POST', 'OPTIONS'], serializer_class=None,
         url_path=r'dataset/?$')
     def dataset(self, request, pk):
         self._object = self.get_object() # force call of check_object_permissions()
-        rq_id = f"import:dataset-for-project.id{pk}-by-{request.user}"
 
         if request.method in {'POST', 'OPTIONS'}:
             return self.import_annotations(
                 request=request,
                 db_obj=self._object,
                 import_func=_import_project_dataset,
                 rq_func=dm.project.import_dataset_as_project,
-                rq_id=rq_id,
+                rq_id_template=self.IMPORT_RQ_ID_TEMPLATE
             )
         else:
             action = request.query_params.get("action", "").lower()
             if action in ("import_status",):
                 queue = django_rq.get_queue(settings.CVAT_QUEUES.IMPORT_DATA.value)
+                rq_id = request.query_params.get('rq_id')
```

Comment:
Shouldn't `rq_id` be validated somehow? What's to prevent a user from checking someone else's request status?

Reply:
Fixed